### DISCUSSION TITLE
Should the website and API match?

### DISCUSSION THREAD

**Discussion by simurgh9** [weather-gov/api#650] [Q&A]
They do for the most part. However, take for example, the station for the coordinates 35.5, -97.6 is KPWA. When I look at,

https://api.weather.gov/stations/KPWA/observations/latest

and

https://forecast.weather.gov/MapClick.php?lat=35.5&lon=-97.6

The picture (icon) and "short forecast" or "text description" never matches.

Is there a way to get what the website gets?

**Comment by dwhitemv25** [comment]
Answered in #427, unless something has changed in the last few years.

**Comment by simurgh9** [comment]
Interesting, so if one wanted the current observations picture and short forecast they could consult the XML api? Is the XML api documented anywhere?

BTW is one more accurate than the other?

**Reply by dwhitemv25** [reply]
Heh, didn't realize the post I linked describes a DWML output option for MapClick. That should be the same data as in the human-readable page. That would be an easy way to get the forecast.weather.gov text forecasts, if you want those specifically. There is also a json output.

There is an XML interface to National Digital Forecast Database at https://graphical.mdl.nws.noaa.gov/xml/. The `/gridpoints` endpoint is a JSON version of the NDFD XML data. There are no text forecasts in there though, just the raw data.

Some light googling didn't find any documentation for MapClick, and if the DWML is correct, the feature is experimental and not operational (like most of the API). I'm not sure how hard they want people beating on that interface.

As far as accuracy goes, it's all from the same data source, just different digestions.
